services/bundle_analysis.py

```diff
@@ -142,3 +155,227 @@ def process_upload(self, upload: Upload) -> ProcessingResult:
+            else:
+                res = await self.repository_service.post_comment(pullid, message)
+                pull.database_pull.bundle_analysis_commentid = res["id"]
+            return True
+        except TorngitClientError:
+            log.error(
+                "Error creating/updapting PR comment",
+                extra=dict(
+                    commitid=self.commit.commitid,
+                    report_key=self.commit_report.external_id,
+                    pullid=pullid,
+                ),
+            )
+            return False
+
+    def _build_message(self, pull: EnrichedPull) -> str:
+        comparison = ComparisonLoader(pull).get_comparison()
+        bundle_changes = comparison.bundle_changes()
+
+        lines = [
+            "## Bundle Report",
+            "",
+        ]
+
+        # total bundle size delta
+        total_size_delta = 0
+        for bundle_change in bundle_changes:
+            total_size_delta += bundle_change.size_delta
+        amount = self._bytes_readable(total_size_delta)
+        if total_size_delta > 0:
+            lines.append(
+                f"Changes will increase total bundle size by {amount} :arrow_up:"
+            )
+        elif total_size_delta < 0:
+            lines.append(
+                f"Changes will decrease total bundle size by {amount} :arrow_down:"
+            )
+        else:
+            lines.append("Changes will not impact bundle size")
+        lines.append("")
+
+        # table of bundles
+        lines += [
+            "| Bundle name | Size | Change |",
+            "| ----------- | ---- | ------ |",
+        ]
+        for bundle_change in bundle_changes:
+            bundle_name = bundle_change.bundle_name
+            head_bundle_report = comparison.head_report.bundle_report(bundle_name)
+            size = self._bytes_readable(head_bundle_report.total_size())
+
+            change_size = bundle_change.size_delta
+            icon = ""
+            if change_size > 0:
+                icon = ":arrow_up:"
+            elif change_size < 0:
+                icon = ":arrow_down:"
+            lines.append(
+                f"| {bundle_change.bundle_name} | {size} | {self._bytes_readable(change_size)} {icon} |"
+            )
+        return "\n".join(lines)
+
+    def _bytes_readable(self, bytes: int) -> str:
+        # TODO: this could maybe be a helper method in `shared`
+
+        bytes = abs(bytes)
+
+        if bytes < 1000:
+            bytes = round(bytes, 2)
+            return f"{bytes} bytes"
+
+        kilobytes = bytes / 1000
+        if kilobytes < 1000:
+            kilobytes = round(kilobytes, 2)
+            return f"{kilobytes}kb"
+
+        megabytes = kilobytes / 1000
+        if megabytes < 1000:
+            megabytes = round(megabytes, 2)
+            return f"{megabytes}mb"
+
+        gigabytes = round(megabytes / 1000, 2)
+        return f"{gigabytes}gb"
```

Comment:
i think it may be better to upper case the characters for the units in this function